File: server/node/src/lib/gateway.js
import braintree from 'braintree';

const { BRAINTREE_MERCHANT_ID, BRAINTREE_PRIVATE_KEY, BRAINTREE_PUBLIC_KEY } =
  process.env;

let gateway;

export function getGateway() {
  if (!gateway) {
    gateway = new braintree.BraintreeGateway({
      environment: braintree.Environment.Sandbox, // use braintree.Environment.Production for production environment
      merchantId: BRAINTREE_MERCHANT_ID,
      privateKey: BRAINTREE_PRIVATE_KEY,
      publicKey: BRAINTREE_PUBLIC_KEY,
    });
  }
  return gateway;
}

File: server/ruby/src/lib/gateway.rb
# frozen_string_literal: true

require 'braintree'

def get_braintree_gateway
  @braintree_gateway ||= Braintree::Gateway.new(
    environment: :sandbox,
    merchant_id: ENV['BRAINTREE_MERCHANT_ID'],
    public_key: ENV['BRAINTREE_PUBLIC_KEY'],
    private_key: ENV['BRAINTREE_PRIVATE_KEY']
  )
end
